server/src/core/link-generator/link-generator-service.ts

```diff
@@ -12,13 +16,21 @@ export class Links implements LinkGeneratorService {
     this.args = args;
   }
 
-  getUserDonateLink(user: User, amount: number): string {
+  async getUserDonateLink(user: User, amount: number): Promise<string> {
     const { name, email, phone } = user;
-    return this.getDonateLink({ name, email, phone, amount });
+    try {
+      const longLink = this.getDonateLink({ name, email, phone, amount });
+      const shortLink = await this.args.bitly.shortenLink(longLink);
+      return shortLink;
+    }
+    catch (e) {
+      rethrowIfAppError(e);
+    }
   }
   getDonateLink(args: DonateLinkArgs): string {
     const { name, email, phone } = args;
-    let amount = args.amount > 2000 ? args.amount : 2000;
-    return `${this.args.baseUrl}?donate=true&n=${name}&e=${email}&p=${phone}&a=${amount}`;
+    const amount = args.amount < 2000 ? 2000 : args.amount;
```

Comment:
I still don't think that the min amount should be enforced in the link generator. I think it should be enforce in the reporting service that calls getUserDonateLink.

Reply:
If it's tedious to change. Then we could just merge this as is.